monocle/types.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (C) 2021 Monocle authors
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Generated by monocle-codegen. DO NOT EDIT!
+from typing import List
+from dataclasses import dataclass
+from dataclasses_json import dataclass_json
+
+
+@dataclass_json
+@dataclass
+class ProjectDefinition:
+    name: str
+    repository_regex: str
+    branch_regex: str
+    file_regex: str
```

Comment:
note: protobuf does not differentiate optional and required values, and a missing attribute is set to its default value, e.g. the empty string here.

Thus the json encoded version may include the default value too, and I think that's acceptable as it makes the webclient easier to implement (because in javascript, "" is not equal to null).

We may want to make the json serialization more compact by omitting the default values in the future, but that will require some more work.